server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -394,6 +412,16 @@ private static ClusterState createFailureStoreIndex(
             return currentState;
         }
 
+        var indexSettings = MetadataRolloverService.HIDDEN_INDEX_SETTINGS;
+        // Optionally set a custom refresh interval for the failure store index.
+        var refreshInterval = getFailureStoreRefreshInterval(settings);
+        if (refreshInterval != null) {
+            indexSettings = Settings.builder()
+                .put(IndexMetadata.SETTING_INDEX_HIDDEN, true)
```

Comment:
I think here we should put `MetadataRolloverService.HIDDEN_INDEX_SETTINGS` and not deconstruct it. Even if now it only has this setting, it's possible that this might change, and this change might not get propagated here.
```suggestion
            indexSettings = Settings.builder()
                .put(MetadataRolloverService.HIDDEN_INDEX_SETTINGS)
```